Sources/CMakeLists.txt

```diff
@@ -6,6 +6,32 @@
 # See http://swift.org/LICENSE.txt for license information
 # See http://swift.org/CONTRIBUTORS.txt for Swift project authors
 
+# cmake generation for Swift adds an order only dependency, which matches how C-family languages
+# works. In that case, however, ninja (and presumably other generators) will rebuild on header
+# changes. That's not the case for Swift, and thus if A -> B, A is not being rebuilt when the
+# ABI/API of B changes.
+#
+# For now workaround this by touching a file whenever B is rebuilt and then compiling that file as
+# part of A. Ideally this file would be generated by each of the targets, but that dependency didn't
+# seem to be being tracked.
+function(add_forced_dependencies TARGET)
+  cmake_parse_arguments(ARGS "" "" "PUBLIC" ${ARGN})
+
+  foreach(DEPENDENCY ${ARGS_PUBLIC})
+    target_link_libraries(${TARGET} PUBLIC
+      ${DEPENDENCY}
+    )
+
+    add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/forced-${DEPENDENCY}-dep.swift
```

Comment:
Then this part can be done in the for-loop.

Reply:
All this is already in the for-loop, or you just mean keep it there? Sure.